document-parsers/langchain4j-document-parser-llamaparse/src/main/java/dev/langchain4j/data/document/parser/llamaparse/LlamaParseClient.java

```diff
@@ -0,0 +1,165 @@
+package dev.langchain4j.data.document.parser.llamaparse;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.MediaType;
+import okhttp3.MultipartBody;
+import okhttp3.OkHttpClient;
+import okhttp3.RequestBody;
+import okhttp3.ResponseBody;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Duration;
+
+import static com.google.gson.FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES;
+
+@Slf4j
+public class LlamaParseClient {
+    private static final Gson GSON = new GsonBuilder()
+            .setFieldNamingPolicy(LOWER_CASE_WITH_UNDERSCORES)
+            .create();
+
+    private final LlmaParseApi llmaParseApi;
+    private String apiKey;
+
+    @Builder
+    public LlamaParseClient(String baseUrl,
+                            Duration timeout,
+                            String apiKey) {
+        this.apiKey = apiKey;
+        OkHttpClient.Builder okHttpClientBuilder = new OkHttpClient.Builder()
+                .addInterceptor(new ApiKeyInsertingInterceptor(apiKey))
+                .callTimeout(timeout)
+                .connectTimeout(timeout)
+                .readTimeout(timeout)
+                .writeTimeout(timeout);
+
+        OkHttpClient okHttpClient = okHttpClientBuilder.build();
+
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl(baseUrl)
```

Comment:
```suggestion
                .baseUrl(getOrDefault(baseUrl, "https://api.cloud.llamaindex.ai/api/parsing/"))
```
Not nice to force the user to figure out what is the value of `baseUrl`